packages/dd-trace/src/heap_snapshots.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+async function scheduleSnapshot (config, total) {
+  if (total > config.heapSnapshot.count) return
+
+  const { setTimeout } = require('timers/promises')
+  const { writeHeapSnapshot } = require('v8')
+
+  await setTimeout(config.heapSnapshot.interval * 1000, null, { ref: false })
+  await clearMemory()
+  writeHeapSnapshot(getName(config.heapSnapshot.folder))
+  await scheduleSnapshot(config, total + 1)
+}
+
+async function clearMemory () {
+  const { setImmediate } = require('timers/promises')
+
+  globalThis.gc()
+  await setImmediate()
+  globalThis.gc() // Run full GC a second time for anything missed in first GC.
```

Comment:
If this is a know issue, I prefer if we reference a bug report/documentation/blog post specifying why this is required

Reply:
I'm rather baffled by the need for this. I don't question it, as I don't have enough expertise in V8 GC, but I'm surprised. I would've expected the V8 heap snapshots either wouldn't include garbage, or if they do, an analysis tool loading them could still identify garbage as objects disconnected from GC roots and ignore it.

Reply:
To be honest, doing a GC here is not really strictly needed anyway. It's just done to remove any memory that can be recollected before the snapshot to avoid noise from normal memory usage as much as possible and make the snapshot smaller. In that sense a single pass would probably be more than enough, but at that point we might as well do 2 for good measure.